pkgs/development/libraries/opencolorio/default.nix

```diff
@@ -1,22 +1,36 @@
-{ stdenv, lib, fetchurl, cmake, unzip, boost }:
+{ stdenv, lib, fetchFromGitHub, cmake, boost, pkgconfig, lcms2, tinyxml, git }:
+
+with lib;
 
 stdenv.mkDerivation rec {
   name = "opencolorio-${version}";
   version = "1.1.0";
 
-  src = fetchurl {
-    url = "https://github.com/imageworks/OpenColorIO/archive/v1.0.9.zip";
-    sha256 = "1vi5pcgj7gv8fp6cdnhszwfh7lh38rl2rk4c5yzsvmgcb7xf48bx";
+  src = fetchFromGitHub {
+    owner = "imageworks";
+    repo = "OpenColorIO";
+    rev = "v${version}";
+    sha256 = "0cjsyn681xsg89lirlll5pqlsqg2vnk1278iiicmzjy2a2v8x7zq";
   };
 
   outputs = [ "bin" "out" "dev" ];
 
-  buildInputs = [ cmake unzip ] ++ lib.optional stdenv.isDarwin boost;
+  # TODO: Investigate whether git can be dropped: It's only used to apply patches
+  nativeBuildInputs = [ cmake pkgconfig git ];
+
+  buildInputs = [ lcms2 tinyxml ] ++ optional stdenv.isDarwin boost;
+
+  CXXFLAGS = "-Wno-error=unused-function -Wno-error=deprecated-declarations";
```

Comment:
Can disable `-Werror` all together? This is a maintenance nightmare when upgrading gcc.

Reply:
And could you also report the `-Werror` issue to uptream and add a comment with a link here.

Reply:
They can set it explicit in CFLAGS when running their CI build: https://github.com/imageworks/OpenColorIO/blob/master/.travis.yml
Apart from that they should default `-Wall`.

Reply:
http://blog.schmorp.de/2016-02-27-tidbits-for-the-love-of-god-dont-use-werror.html

Reply:
Looks like there's already a PR: https://github.com/imageworks/OpenColorIO/pull/551 and some discussion: https://github.com/imageworks/OpenColorIO/pull/518 on the subject of Werror. Not clear if/when they'll get round to doing something about it though.

For completeness, I opened an issue as well: https://github.com/imageworks/OpenColorIO/issues/559

Reply:
I've replaced the CXX_FLAGS expression which disabled errors on unused-function and deprecated-declarations with a postPatch that strips both occurences of -Werror.